File: Homemake.Scripting.Languages.In.Two.Weeks/python/chap03/ply/LexerRun.py
import sys
from stone.Lexer import Lexer
from stone.Token import Token

def main():
    file_path = "test.txt"
    if len(sys.argv) > 1:
        file_path = sys.argv[1]

    with open(file_path) as f:
        l = Lexer(f)
        t = l.read()
        while t != Token.EOF:
            print("=> " + t.getText())
            t = l.read()

main()

File: Language.Implementation.Patterns/cxx/ch02_basic_parsing_patterns/p03_recursive_descent/run/LexerRun.cpp
#include <string>
#include <iostream>
#include <fstream>
#include <sstream>

#include "Token.hpp"
#include "ListLexer.hpp"
#include "read_file.hpp"

int main(int argc, char* argv[]) {
    const char* file_path = "test.txt";

    if (argc == 2) {
        file_path = argv[1];
    }

    ListLexer lexer{read_file(file_path)};
    auto t = lexer.nextToken();
    while (t.type != Lexer::EOF_TYPE) {
        std::cout << "=> " << t.toString(lexer) << "\n";
        t = lexer.nextToken();
    }
}
